app/gui/src/dashboard/layouts/Settings/Input.tsx

```diff
@@ -27,31 +29,55 @@ export default function SettingsInput<T extends Record<keyof T, string>>(
     hidden: hiddenRaw,
     editable,
     descriptionId,
+    comboBoxProps,
     type = 'text',
   } = data
   const { getText } = useText()
 
   const isEditable = typeof editable === 'function' ? editable(context) : (editable ?? true)
   const hidden = typeof hiddenRaw === 'function' ? hiddenRaw(context) : (hiddenRaw ?? false)
 
-  const Input = INPUT_TYPE_MAP[type]
+  switch (type) {
+    case 'comboBox': {
+      const extraProps =
+        typeof comboBoxProps === 'function' ? comboBoxProps(context) : comboBoxProps
+      if (!extraProps) {
+        // eslint-disable-next-line no-restricted-properties
+        console.error('Settings inputs of type `comboBox` MUST be passed `comboBoxProps`.')
```

Comment:
I find this error message confusing. Can we express the same thing on the type level? Also, even if I pass `comboBoxProps`, it doesn’t mean I won’t see this error.